components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionState.kt

```diff
@@ -46,5 +48,27 @@ class GeckoEngineSessionState internal constructor(
         } catch (e: JSONException) {
             GeckoEngineSessionState(null)
         }
+
+        /**
+         * Creates a [GeckoEngineSessionState] from the given [JsonReader].
+         */
+        fun from(reader: JsonReader): GeckoEngineSessionState = try {
+            reader.beginObject()
+
+            val key = reader.nextName()
+            if (key != GECKO_STATE_KEY) {
+                throw AssertionError("Unknown state key: $key")
```

Comment:
Instead of throwing, should we just discard here and return `GeckoEngineSessionState(null)`? Otherwise switching `engine` would crash when reading state from the old engine?

Reply:
Ah, interesting. That makes sense. I guess I would have to discard until I hit the matching END_OBJECT token.